### DISCUSSION TITLE
Access to download workflow run logs API

### DISCUSSION THREAD

**Discussion by AllanChain** [community/community#24742] [Apps, API and Webhooks]
<p>I am trying to get workflow run logs in other repos.</p>
<p>The <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#download-workflow-run-logs" rel="noopener nofollow ugc">docs</a> says:</p>
<blockquote>
<p>Anyone with read access to the repository can use this endpoint.</p>
</blockquote>
<p>However, either with or without my personal access token, I am getting this message:</p>
<pre><code class="lang-json">{
  "message": "Must have admin rights to Repository.",
  "documentation_url": "https://docs.github.com/rest/reference/actions#download-workflow-run-logs"
}
</code></pre>
<p>Do I misunderstand “Anyone with read access”? Or is the doc outdated?</p>

**Comment by justanotheranonymoususer** [comment]
<p>I have exactly the same problem. Perhaps it’s a temporary issue? Happens for me with the example from the docs as well:</p>
<pre><code class="lang-auto">curl \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/octocat/hello-world/actions/runs/42/logs
</code></pre>

**Comment by justanotheranonymoususer** [comment]
<p>What I also found is that you can go to the job URL and query the sha, job id parameters to then get the job log as said here:</p><aside class="quote" data-post="1" data-topic="18227">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/any-way-to-determine-url-to-raw-logs-of-github-actions/18227">Any way to determine url to raw logs of github actions?</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Hello, 
I would like to provide link and possible line location of raw log file url that is found in the “View raw logs” pull down menu on the Actions  page. 
Is it possible to assible this link from GITHUB env variables during the action run?
  </blockquote>
</aside>

**Comment by justanotheranonymoususer** [comment]
<p>Bad news, this method only works if a user is logged in in the browser. What a disaster, why did the GitHub devs made it so clumsy?!</p>

**Comment by AllanChain** [comment, marked as answer]
<p>Fixed this using a token with repo scope. Maybe GitHub should improve the error message and the docs.</p>

**Comment by NotNite** [comment]
This is still an issue in 2025 - the docs **still** claim "Anyone with read access to the repository can use this endpoint", but actually trying it will get you the admin rights error. The `github.com/:owner/:repo/commit/:sha/checks/:id/logs` endpoint requires a session from the browser, so there doesn't seem to be a way to read logs for a workflow from code unless you are an admin of the repository. :(
